File: services/academic-assessment-service/src/test/java/com/visor/school/assessment/model/AssessmentStatusTest.java
package com.visor.school.assessment.model;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class AssessmentStatusTest {

    @Test
    void shouldHaveAllRequiredAssessmentStatusValues() {
        AssessmentStatus[] statuses = AssessmentStatus.values();

        assertTrue(java.util.Arrays.asList(statuses).contains(AssessmentStatus.DRAFT));
        assertTrue(java.util.Arrays.asList(statuses).contains(AssessmentStatus.PUBLISHED));
        assertTrue(java.util.Arrays.asList(statuses).contains(AssessmentStatus.GRADING));
        assertTrue(java.util.Arrays.asList(statuses).contains(AssessmentStatus.COMPLETED));
        assertEquals(4, statuses.length);
    }

    @Test
    void shouldHaveCorrectStatusNames() {
        assertEquals("DRAFT", AssessmentStatus.DRAFT.name());
        assertEquals("PUBLISHED", AssessmentStatus.PUBLISHED.name());
        assertEquals("GRADING", AssessmentStatus.GRADING.name());
        assertEquals("COMPLETED", AssessmentStatus.COMPLETED.name());
    }
}

File: services/academic-assessment-service/src/test/kotlin/com/visor/school/assessment/model/AssessmentStatusTest.kt
package com.visor.school.assessment.model

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class AssessmentStatusTest {

    @Test
    fun `should have all required assessment status values`() {
        val statuses = AssessmentStatus.values()

        assertTrue(statuses.contains(AssessmentStatus.DRAFT))
        assertTrue(statuses.contains(AssessmentStatus.PUBLISHED))
        assertTrue(statuses.contains(AssessmentStatus.GRADING))
        assertTrue(statuses.contains(AssessmentStatus.COMPLETED))
        assertEquals(4, statuses.size)
    }

    @Test
    fun `should have correct status names`() {
        assertEquals("DRAFT", AssessmentStatus.DRAFT.name)
        assertEquals("PUBLISHED", AssessmentStatus.PUBLISHED.name)
        assertEquals("GRADING", AssessmentStatus.GRADING.name)
        assertEquals("COMPLETED", AssessmentStatus.COMPLETED.name)
    }
}

